### ISSUE TITLE
S3 Content-Length issue with AWS Go SDK

### ISSUE THREAD

**Issue by applike-ss** [localstack/localstack#3833]
# Type of request: This is a bug

[x] bug report

# Detailed description
When adding a file to a bucket and then reading it via to go sdk, i do get a content-length of 0, while the entries body is actually filled and when reading it out has the expected length. This is since v0.11.5, v0.11.4 did work

## Expected behavior
ContentLength property should be filled in aws go sdk

## Actual behavior
ContentLength property is nil

# Steps to reproduce
* start localstack (>=v0.11.5) via docker and expose it
* copy a file to a new localstack s3 bucket
* fetch it via the aws go sdk and check the output content length

v0.11.4:
![image](https://user-images.githubusercontent.com/31846129/113714228-78e84880-96e8-11eb-916f-2c0b9173ad7f.png)

v0.11.5 and above (tested until v0.12.9):
![image](https://user-images.githubusercontent.com/31846129/113714078-4dfdf480-96e8-11eb-870d-6d3afee9d248.png)

As you can see there are some headers omitted.

## Command used to start LocalStack
docker run localstack/localstack:0.11.4
docker run localstack/localstack:0.11.5

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
none provided

**Comment by duongpv7** [comment]
Hi @applike-ss, can you share your Go code to help us reproducing this issue?
Thanks

**Comment by applike-ss** [comment]
Just wanted to try my test-case, now getting this :see_no_evil: 
```
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-05-06T06:06:24:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-05-06T06:06:34:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-05-06T06:06:48:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
2021-05-06T06:07:02:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
2021-05-06T06:07:09:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
Waiting for all LocalStack services to be ready
2021-05-06T06:07:17:WARNING:localstack.services.plugins: Service "s3" not yet available, retrying...
```
i will report back my test case code, once it get my localstack working again

**Comment by applike-ss** [comment]
Ok, so i did now test 0.11.4 against 0.11.5 because my 0.12 versions suddenly don't want to start the s3 mock anymore. However feel free to test against the most recent version if you can.
Here's my test go code:
```go
func main() {
	config := &aws.Config{
		CredentialsChainVerboseErrors: aws.Bool(true),
		MaxRetries:                    aws.Int(30),
		Region:                        aws.String(endpoints.EuCentral1RegionID),
		Endpoint:                      aws.String("http://127.0.0.1:4566"),
		S3ForcePathStyle:              aws.Bool(true),
	}

	newSession, err := session.NewSession(config)

	if err != nil {
		panic(err)
	}
	client := s3.New(newSession)

	bucketName:="my-bucket"

	cbi := s3.CreateBucketInput{
		ACL:                        aws.String("public"),
		Bucket:                     aws.String(bucketName),
	}
	_, err = client.CreateBucket(&cbi)
	if err != nil {
		if awsErr, ok := err.(awserr.RequestFailure);ok {
			code := awsErr.Code()
			if code !="BucketAlreadyExists" {
				fmt.Println(err.Error())
			}
		} else {
			fmt.Println(err.Error())
		}
	}
	
	reader := strings.NewReader("foobarbaz")
	filename := "my-file.txt"

	poi := s3.PutObjectInput{
		ACL:                       aws.String("public-read"),
		Body:                      reader,
		Key: aws.String(filename),
		Bucket:                    aws.String(bucketName),
	}

	_, err=client.PutObject(&poi)
	if err != nil {
		fmt.Println(err.Error())
	}

	goi := s3.GetObjectInput{
		Bucket:                    aws.String(bucketName),
		Key:                        aws.String(filename),
	}

	goo, err := client.GetObject(&goi)
	if err != nil {
		fmt.Println(err.Error())
	}

	fmt.Println(fmt.Sprintf("content length is: %v", goo.ContentLength))
}
```
When run, it will either report:
`content length is: nil` for versions that don't send the content lenth correctly or `content length is: 0x.....` for versions that send the content length correctly.

**Comment by whummer** [comment]
Thanks @applike-ss @duongpv7 . This should be fixed in #3992 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by applike-ss** [comment]
```
LocalStack version: 0.12.11
LocalStack Docker container id: b9f0ad44f816
LocalStack build date: 2021-05-11
LocalStack build git hash: e79d7bf6
```
Works for me, thanks! @whummer @duongpv7
